lib/telemetry/metrics.go

```diff
@@ -0,0 +1,188 @@
+package telemetry
```

Comment:
curious why we're implementing all these instead of just using the underlying opentelemetry tools -- some add functionality but others appear to be largely passthroughs